redfish-core/lib/log_services.hpp

```diff
@@ -6606,4 +6613,613 @@ inline void requestRoutesSystemHardwareIsolationLogService(App& app)
 }
 #endif // BMCWEB_ENABLE_HW_ISOLATION
 
+#ifdef BMCWEB_ENABLE_LINUX_AUDIT_EVENTS
+/****************************************************
+ * Redfish AuditLog interfaces
+ ******************************************************/
+inline void handleLogServicesAuditLogGet(
+    crow::App& app, const crow::Request& req,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& systemName)
+{
+    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    {
+        return;
+    }
+    if (systemName != "system")
+    {
+        messages::resourceNotFound(asyncResp->res, "ComputerSystem",
+                                   systemName);
+
+        return;
+    }
+    asyncResp->res.jsonValue["@odata.id"] =
+        "/redfish/v1/Systems/system/LogServices/AuditLog";
+    asyncResp->res.jsonValue["@odata.type"] = "#LogService.v1_2_0.LogService";
+    asyncResp->res.jsonValue["Name"] = "Audit Log Service";
+    asyncResp->res.jsonValue["Description"] = "Audit Log Service";
+    asyncResp->res.jsonValue["Id"] = "AuditLog";
+    asyncResp->res.jsonValue["OverWritePolicy"] = "WrapsWhenFull";
+    asyncResp->res.jsonValue["Entries"]["@odata.id"] =
+        "/redfish/v1/Systems/system/LogServices/AuditLog/Entries";
+
+    std::pair<std::string, std::string> redfishDateTimeOffset =
+        redfish::time_utils::getDateTimeOffsetNow();
+    asyncResp->res.jsonValue["DateTime"] = redfishDateTimeOffset.first;
+    asyncResp->res.jsonValue["DateTimeLocalOffset"] =
+        redfishDateTimeOffset.second;
+}
+
+inline void requestRoutesAuditLogService(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/<str>/LogServices/AuditLog/")
+        .privileges(redfish::privileges::getLogService)
+        .methods(boost::beast::http::verb::get)(
+            std::bind_front(handleLogServicesAuditLogGet, std::ref(app)));
+}
+
+static LogParseError
+    fillAuditLogEntryJson(const nlohmann::json& auditEntry,
+                          nlohmann::json::object_t& logEntryJson)
+{
+    if (auditEntry.is_discarded())
+    {
+        return LogParseError::parseFailed;
+    }
+
+    std::string logEntryID;
+    std::string entryTimeStr;
+    std::string messageID;
+    nlohmann::json messageArgs = nlohmann::json::array();
+    for (const auto& item : auditEntry.items())
+    {
+        if (item.key() == "ID")
+        {
+            logEntryID = item.value();
+        }
+
```

Comment:
I think it would be better to remove the blank line here.

Reply:
Addressed in 1110 rebase.